lib/app_profiler/middleware/upload_action.rb

```diff
@@ -34,7 +34,11 @@ def append_headers(response, upload:, autoredirect:)
         end
 
         def profile_url(upload)
-          "#{AppProfiler.speedscope_host}#profileURL=#{upload.url}"
+          if AppProfiler.profile_url_formatter.respond_to?(:call)
```

Comment:
I'd just do a `nil` check. You expect that property to be a callable, if someone provide something else, it's better if it breaks loudly than to be ignored silently.